Tc drlk 2 1

aabgd opened the pull request:
#### Testing

This PR updates Test_TC_DRLK_2_1.yaml and Fixes [#492](https://github.com/project-chip/matter-test-scripts/issues/492)

Test specification is [TC-DRLK-2.1](https://github.com/aabgd/chip-test-plans/blob/master/src/cluster/door_lock_Cluster.adoc#411-tc-drlk-21-attributes-check-dut---server)

Updated:

- steps 5c and 6c
- ci-pics-values
- Set maxValue to 255 in steps 11a, 12a and 13a
- Corrected step 31d
- Added missing step 32d
- Updated values for steps 40b, 42b
- Updated PICs for steps 5, 6, 7, 20, 25, 26, 28, 30 and 34:
  - Left disabled:
```
    - DRLK.S.A0022=0
    - DRLK.S.A0028=0
    - DRLK.S.A0032=0
    - DRLK.S.A0035=0
    - DRLK.S.M.LocalProgrammingFeaturesAttributeWritable=0
```
  - Left enabled: (you should enabled these values in ZAP tool for the lock-app at endpoint 1 to test locally)
```
    - DRLK.S.A0004=1
    - DRLK.S.A0005=1
    - DRLK.S.A0006=1
    - DRLK.S.A0027=1
    - DRLK.S.A002a=1
    - DRLK.S.A002c=1
```

Testing:
In one terminal:
Build and run the lock-app:
```
$ ./scripts/build/build_examples.py --target darwin-arm64-lock-no-ble build
$ ./out/darwin-arm64-lock-no-ble/chip-lock-app
```

In a second terminal:
Run Test:
```
$ ./out/darwin-arm64-chip-tool/chip-tool pairing code 0x12344321 MT:-24J0AFN00KA0648G00
$ ./scripts/tests/chipyaml/chiptool.py tests Test_TC_DRLK_2_1 --server_path ./out/darwin-arm64-chip-tool/chip-tool
```

aabgd committed:
changed expected values from 5c and 6c to 0

aabgd committed:
Set to 1 the following PICS:
DRLK.S.A0004=1
DRLK.S.A0005=1
DRLK.S.A0006=1

aabgd committed:
Set maxValue to 255 in steps 11a, 12a and 13a

aabgd committed:
updated ci-pics-values, corrected step 31d, added missing step 32d

aabgd committed:
updated values for 40b, 42b, and updated PICs for steps 42a/b/c, 43a/b/c and 44a/b/c

juandediosg reviewed (commented):
It would be good practice to include summary, scope, related issues/PRs, and testing details in the PR description.

aabgd committed:
Finally enabled the following PICs in both ZAP tool for lock-app and ci-pics-values
DRLK.S.A0004=1
DRLK.S.A0005=1
DRLK.S.A0006=1
DRLK.S.A0027=1
DRLK.S.A002a=1
DRLK.S.A002c=1
and left the following PICs disabled
DRLK.S.A0022=0
DRLK.S.A0028=0
DRLK.S.A0032=0
DRLK.S.A0035=0
DRLK.S.M.LocalProgrammingFeaturesAttributeWritable=0

aabgd commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -1403,6 +1405,28 @@ tests:
+    - label: "Step 32d: TH writes UserCodeTemporaryDisableTime attribute as 0"
+      PICS:
+          " ( DRLK.S.F00 || DRLK.S.F01 ) && DRLK.S.A0031 &&
+          DRLK.S.M.UserCodedTemporaryDisableTimeAttributeWritable "
```
Not needed

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Gated steps 5a, 5b, 6a and 6b for DoorOpenEvents and DoorClosedEvents

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

cecille commented on examples/lock-app/lock-common/lock-app.zap:
```diff
@@ -5372,6 +5372,54 @@
               "reportableChange": 0
             },
+            {
+              "name": "DoorOpenEvents",
```
Is this now just mocking a device that never has the door opened? Is there any test that door open events does anything?

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -299,13 +299,29 @@ tests:
               maxValue: 4294967294
 
+    - label: "Step 5a: TH reads DoorOpenEvents attribute from DUT"
+      PICS: DRLK.S.F05 && !DRLK.S.A0004
```
```suggestion
      PICS: !DRLK.S.A0004
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -321,7 +337,7 @@ tests:
           ./chip-tool doorlock read door-open-events 1 1
 
-          Via the TH (chip-tool), verify that the DoorOpenEvents attribute value as 10.
+          Via the TH (chip-tool), verify that the DoorOpenEvents attribute value is 0.
```
Can this be updated to be automated? I'm not sure why this is a manual step currently, but the point of the issue was to automate the test fully. The statement in the comment above is incorrect - if the attribute is present, the value is required to be 0 and it looks like someone just made this test pass by basically commenting out this failing step.

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Update src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml

Co-authored-by: C Freeman <cecille@google.com>

aabgd committed:
automated steps 5c and 6c

aabgd committed:
added steps 5d, 5e, 6d and 6e

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Fix review comments: revert maxValue to 253 and fix PICS logic

- Steps 11a, 12a, 13a: Revert maxValue from 255 to 253 (0xFD) as per spec
- Step 30g: Negate LocalProgrammingFeaturesAttributeWritable PICS so
  UNSUPPORTED_WRITE is only expected when attribute is NOT writable

aabgd commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -1273,7 +1289,7 @@ tests:
           "Step 30g: TH sends a write message to change any one of the bits of
           the LocalProgrammingFeatures."
-      PICS: DRLK.S.A002c
+      PICS: DRLK.S.A002c && DRLK.S.M.LocalProgrammingFeaturesAttributeWritable
```
Done. Thank you!

aabgd commented on examples/lock-app/lock-common/lock-app.zap:
```diff
@@ -5372,6 +5372,54 @@
               "reportableChange": 0
             },
+            {
+              "name": "DoorOpenEvents",
```
The [Issue #2863](https://github.com/CHIP-Specifications/chip-test-plans/issues/2863) reported incorrect expected values in steps 5c and 6c (10 and 11 instead of 0 after writing 0). My changes fix this bug.

However, I understand your concern that these steps don't test actual DPS (Door Position Sensor) functionality. The DPS feature bit (0x20) is not enabled in lock-app's FeatureMap (0x000F), and there's no auto-increment logic when door state changes. These steps only verify basic read/write attribute operations.
Regarding tests, there are no other tests for DoorOpenEvents/DoorClosedEvents.

Should I keep these steps as they verify attributes are accessible (basic conformance), or remove steps 5c-5e and 6c-6e (since DPS is not implemented), or would you like me to convert these steps to manual for future DPS implementations?

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
[lock-app] Remove DPS attributes without DPS feature support

Disabled DoorOpenEvents, DoorClosedEvents, and OpenPeriod attributes
as lock-app does not implement DPS feature. Updated test and PICS accordingly.

aabgd committed:
Merge branch 'master' into tc_drlk_2_1

aabgd committed:
Fixed PICS conditions for DPS attributes in steps 5x, 6x, 7x

- Added DRLK.S.F05 feature check to all DoorOpenEvents, DoorClosedEvents and OpenPeriod test cases
- Added negative test cases for unsupported attributes

aabgd committed:
Fixed steps 5x, 6x and 7x

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -353,51 +346,42 @@ tests:
               maxValue: 4294967294
 
+    - label: "Step 6a: TH reads DoorClosedEvents attribute from DUT"
+      PICS: DRLK.S.F05 && !DRLK.S.A0005
```
```suggestion
      PICS: !DRLK.S.A0005
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -353,51 +346,42 @@ tests:
+
     - label: "Step 6b: TH writes DoorClosedEvents attribute as 0"
-      PICS: DRLK.S.A0005
+      PICS: DRLK.S.F05 && DRLK.S.A0005
```
```suggestion
      PICS: DRLK.S.A0005
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -353,51 +346,42 @@ tests:
+          value: 0
+
+    - label: "Step 6c: TH reads DoorClosedEvents attribute from DUT"
+      PICS: DRLK.S.F05 && !DRLK.S.A0005
```
```suggestion
      PICS: !DRLK.S.A0005
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -409,20 +393,43 @@ tests:
+          error: UNSUPPORTED_ATTRIBUTE
+
+    - label: "Step 7b: TH writes OpenPeriod attribute as 2 minutes"
+      PICS: DRLK.S.F05 && DRLK.S.A0006
```
```suggestion
      PICS: DRLK.S.A0006
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRLK_2_1.yaml:
```diff
@@ -409,20 +393,43 @@ tests:
 
     - label: "Step 7c: TH reads OpenPeriod attribute from DUT"
-      PICS: DRLK.S.A0006
+      PICS: DRLK.S.F05 && DRLK.S.A0006
```
```suggestion
      PICS: DRLK.S.A0006
```

khodya committed:
applying suggestions

khodya committed:
Merge branch 'master' into tc_drlk_2_1

khodya committed:
quotes
